src/libvirtApi/domain.js

```diff
@@ -613,17 +604,11 @@ export function domainInstall({ onAddErrorNotification, vm }) {
     const args = JSON.stringify({
         autostart: vm.autostart,
```

Comment:
Is this option necessarry? I believed this also configured by virt-install --reinstall. I just tried deleting it from the python script, then created VM, changed status of autostart, installed the VM and the correct status of autostart persisted.

Reply:
I indeed verified manually as well that is option is also propagated.

Reply:
Honestly however, to me it seems like a side effect. As it's not part of the VM xml definition.

Reply:
It's true that VM's autostart configuration, is part not part of XML, but it's part of some flags/boolean values which libvirt keeps track for existing VMs.

Before this PR, we lost the autostart configuration, because for install process we undefined and then defined again the VM, so wherever libvirt kept the state of autostart, it was removed with VM undefining.
```
event 'lifecycle' for domain 'centos': Defined Added
event 'lifecycle' for domain 'centos': Undefined Removed
event 'agent-lifecycle' for domain 'centos': state: 'disconnected' reason: 'domain started'
event 'lifecycle' for domain 'centos': Resumed Unpaused
event 'lifecycle' for domain 'centos': Started Booted
event 'lifecycle' for domain 'centos': Defined Added
event 'graphics' for domain 'centos': connect local[IPv4 127.0.0.1 5900] remote[IPv4 127.0.0.1 40200] none
event 'graphics' for domain 'centos': initialize local[IPv4 127.0.0.1 5900] remote[IPv4 127.0.0.1 40200] none
```

With this PR, it seems that virt-install is only pausing the VM for installation. That means that VM with all its non-XML configuration seems to persist:
```
event 'lifecycle' for domain 'centos': Defined Added
event 'agent-lifecycle' for domain 'centos': state: 'disconnected' reason: 'domain started'
event 'lifecycle' for domain 'centos': Resumed Unpaused
event 'lifecycle' for domain 'centos': Started Booted
event 'lifecycle' for domain 'centos': Defined Updated
event 'graphics' for domain 'centos': connect local[IPv4 127.0.0.1 5900] remote[IPv4 127.0.0.1 40198] none
event 'graphics' for domain 'centos': initialize local[IPv4 127.0.0.1 5900] remote[IPv4 127.0.0.1 40198] none
```

So my point is, If virt-xml --reinstall guarantees that VM will not get undefined/redefined in the process, then I think it's not a side effect but expected behavior.

Reply:
This is convincing. I dropped it from the install_machine.py script and domain.js. `testConfigureBeforeInstall` covers this and is still happy.

Reply:
Thanks. Since we still keep the test, if in the future it will start failing, at least we will know that something broke in virt-install and not with us.